src/main/java/fr/mossroy/tomcat/tools/ExternalPropertySource.java

```diff
@@ -47,7 +50,20 @@ public ExternalPropertySource() {
             if (externalPropertiesFile == null || externalPropertiesFile.isEmpty()) {
                 throw new IOException("The external property file location is not set in " + CATALINA_PROPERTIES + " (expected value for " + CATALINA_PROPERTIES_FILE_PROPERTY + ")");
             }
-            FileInputStream fileInputStream = new FileInputStream(externalPropertiesFile);
+            
+            // Replace the Java variables

```

Comment:
The comment is not meaningful enough IMHO. I had to read the code to understand
```suggestion
            // Replace the ${...} placeholders in the externalPropertiesFile String by the corresponding Java properties (when they exist)

```